job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -384,7 +404,12 @@ class WebApi(system: ActorSystem,
                 val resultFuture = jobInfo ? GetJobResult(jobId)
                 resultFuture.map {
                   case JobResult(_, result) =>
-                    ctx.complete(jobReport ++ resultToTable(result))
+                    result match {
+                      case s: Stream[_] =>
```

Comment:
Streams are memorized, so this can be called multiple times. It wouldn't be desirable to keep a large Stream in memory, though. Once persistence of results is implemented it would be better to stream it out of storage again.